zebra-rpc/src/methods/tests/snapshot.rs

```diff
@@ -316,6 +322,88 @@ async fn test_rpc_response_data_for_network(network: Network) {
     snapshot_rpc_getaddressutxos(get_address_utxos, &settings);
 }
 
+async fn test_mocked_rpc_response_data_for_network(network: Network) {
+    // Prepare the test harness.
+
+    let mut settings = insta::Settings::clone_current();
+    settings.set_snapshot_suffix(network_string(network));
+
+    let (latest_chain_tip, _) = MockChainTip::new();
+    let mut state = MockService::build().for_unit_tests();
+    let mempool = MockService::build().for_unit_tests();
+
+    let (rpc, _) = RpcImpl::new(
+        "RPC test",
+        "/Zebra:RPC test/",
+        network,
+        false,
+        true,
+        Buffer::new(mempool, 1),
+        state.clone(),
+        latest_chain_tip,
+    );
+
+    // Test the response format from `z_getsubtreesbyindex` for Sapling.
+
+    // Mock the data for the response.
+    let mut subtrees = BTreeMap::new();
+    let subtree_root = [0u8; 32].as_slice().try_into().expect(
```

Comment:
Nitpick: If we use a fake root value that isn't symmetrical, then this test will catch bugs like #7466.

For example:
```rust
let mut subtree_root = [0u8; 32];
subtree_root[0] = 0xff;
```

Feel free to submit this in another PR, but I don't think it's a blocker because one of the other tests should cover it.

Reply:
PR #7515 already covers this issue, so there's no need to open another PR.